pkg/alertmanager/alertmanager.go

```diff
@@ -97,6 +104,10 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 		stop:      make(chan struct{}),
 		active:    false,
 		activeMtx: sync.Mutex{},
+		configHashMetric: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
Does this metric get exposed? The input `reg` is not the exposed one, but an internal one added to  `alertmanagerMetrics`, where the metric is not exposed. Am I missing anything? Could we just add an assertion to an existing integration test? You could just assert on the count of this metric, for simplicity.

Reply:
You're right. I've added the tests now.